pkg/query-service/integrations/signozio/dynamic_config.go

```diff
@@ -0,0 +1,71 @@
+package signozio
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"go.signoz.io/signoz/ee/query-service/model"
+	"go.signoz.io/signoz/pkg/query-service/constants"
+	"go.uber.org/zap"
+)
+
+var C *Client
+
+const (
+	POST             = "POST"
+	APPLICATION_JSON = "application/json"
+)
+
+type Client struct {
+	Prefix string
+}
+
+func New() *Client {
+	return &Client{
+		Prefix: constants.ConfigSignozIo,
+	}
+}
+
+func init() {
+	C = New()
+}
+
+// FetchDynamicConfigs fetches configs from config.signoz.io
+func FetchDynamicConfigs() (map[string]Config, *model.ApiError) {
+
+	httpResponse, err := http.Get(C.Prefix + "/configs")
+
+	if err != nil {
+		zap.S().Errorf("failed to connect to config.signoz.io", err)
+		return nil, model.BadRequest(fmt.Errorf("unable to connect with config.signoz.io, please check your network connection"))
```

Comment:
@makeavish rather pass a default config when connection to upstream fails.

Reply:
Done